src/main/java/com/openshift/internal/restclient/model/BuildConfig.java

```diff
@@ -138,17 +140,22 @@ public void addBuildTrigger(IBuildTrigger trigger) {
 			triggerNode.get(getPath(BUILD_CONFIG_WEBHOOK_GITHUB_SECRET)).set(github.getSecret());
 			break;
 		case BuildTriggerType.imageChange:
-		case BuildTriggerType.IMAGE_CHANGE:
+		case BuildTriggerType.IMAGE_CHANGE:{
 			if(!(trigger instanceof IImageChangeTrigger)) {
 				throw new IllegalArgumentException("IBuildTrigger of type imageChange does not implement IImageChangeTrigger");
 			}
 			IImageChangeTrigger image = (IImageChangeTrigger)trigger;
-			triggerNode.get(getPath(BUILD_CONFIG_IMAGECHANGE_IMAGE)).set(defaultIfNull(image.getImage()));
-			triggerNode.get(getPath(BUILD_CONFIG_IMAGECHANGE_NAME)).set(defaultIfNull(image.getFrom()));
-			triggerNode.get(getPath(BUILD_CONFIG_IMAGECHANGE_TAG)).set(StringUtils.defaultIfBlank(image.getTag(), ""));
+			if(image.getImage() != null)
+				triggerNode.get(getPath(BUILD_CONFIG_IMAGECHANGE_IMAGE)).set(defaultIfNull(image.getImage()));
```

Comment:
just nitpicking here: don't need defaulIfNull since null check is already done :-)

Same goes for the next lines of code